resources/js/components/Checkout/Login.vue

```diff
@@ -50,24 +52,24 @@ export default {
                 })
             } else if (this.email) {
                 await this.checkEmailAvailability()
+
+                if (this.emailAvailable) {
+                    this.$root.guestEmail = this.email
+                    this.$root.checkout.step = this.nextStep
+                } else {
+                    this.$nextTick(function () {
+                        this.$scopedSlots.default()[0].context.$refs.password.focus()
+                    })
+                }
             } else {
                 Notify(window.config.translations.account.email, 'error')
             }
         },
 
         async checkEmailAvailability() {
-            let responseData = await window.magentoAPI('post', 'customers/isEmailAvailable', {
+            this.emailAvailable = await window.magentoAPI('post', 'customers/isEmailAvailable', {
```

Comment:
With these changes it might make sense to change the function name and return the value we get from Magento's API instead of setting a value.
As we no longer do any checking here, rather returning the result of `isEmailAvailable`